eng/scripts/typespec_build.ps1

```diff
@@ -0,0 +1,110 @@
+#Requires -Version 7.0
+param(
+    [string]$filter,
+    [switch]$clean,
+    [switch]$vet,
+    [switch]$generate,
+    [switch]$skipBuild,
+    [switch]$cleanGenerated,
+    [switch]$format,
+    [switch]$tidy,
+    [string]$emitterOptions = "module-version=0.1.0", # default options
+    [string]$localSpecRepo
+)
+
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+function Process-TypeSpec ()
```

Comment:
Will this work for data plane scenarios? 
@gracewilcox could you review if this script aligns with what commands you run for local developing purpose?

Reply:
For local development, I always run `go generate` which executes the commands in my [build.go](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/security/keyvault/azkeys/build.go) file.

For example:
tsp-client update
go run ./internal/transforms.go
goimports -w .

Reply:
@JiaqiZhang-Dev pls ensure data plane cases work with the new changes in SDK generation.